File: OdonSysBackEnd/Resource/Contract/Contract.Admin/Roles/RoleModel.cs
using Contract.Admin.Users;

namespace Contract.Admin.Roles
{
    public class RoleModel
    {
        public string Name { get; set; }
        public string Code { get; set; }
        public IEnumerable<string> RolePermissions { get; set; }
        public IEnumerable<DoctorModel> UserRoles { get; set; }
        public DateTime DateCreated { get; set; }
        public DateTime DateModified { get; set; }
        public string UserCreated { get; set; }
        public string UserUpdated { get; set; }
    }
}

File: OdonSys-SPA/src/app/core/models/view/role-model.ts
import { DoctorModel } from "./doctor-model";

export class RoleModel {
  constructor(
    public name: string,
    public code: string,
    public userCreated: string,
    public userUpdated: string,
    public dateCreated: Date,
    public dateModified: Date,
    public rolePermissions: string[],
    public userRoles?: DoctorModel[]
  ) { }
}
